libs/model/src/aggregates/user/signIn/utils.ts

```diff
@@ -95,6 +97,32 @@ export async function findUserBySso(
   return users[0];
 }
 
+/*
+ Check minimum balance requirement for new or existing users
+ signing in with native wallets that are below SocialVerified tier
+*/
+async function checkNativeWalletBalance(
+  address: string,
+  foundUser: UserAttributes | null,
+  ethChainId?: number,
+): Promise<UserTierMap> {
+  if (
+    (foundUser?.tier || UserTierMap.IncompleteUser) < UserTierMap.SocialVerified
+  ) {
+    const balances = ethChainId
+      ? await tokenBalanceCache.getBalances({
+          addresses: [address],
+          balanceSourceType: BalanceSourceType.ETHNative,
+          sourceOptions: { evmChainId: ethChainId },
+        })
+      : { [address]: '0' };
+    const balance = BigInt(balances[address] || '0');
+    const minBalance = BigInt(config.TIER.SOCIAL_VERIFIED_MIN_ETH * 1e18);
+    if (balance >= minBalance) return UserTierMap.SocialVerified;
+  }
+  return UserTierMap.NewlyVerifiedWallet;
```

Comment:
We should return the existing tier - won't this downgrade users from `UserTierMap.VerifiedWallet` if they don't have Privy user id set and they sign in after 1 week?